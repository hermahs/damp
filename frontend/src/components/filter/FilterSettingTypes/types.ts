import { IFilterData } from "../../../store/filterStore";

export interface IFilterSettingTypeProp {
    setFilterData: (data: IFilterData) => void;
}