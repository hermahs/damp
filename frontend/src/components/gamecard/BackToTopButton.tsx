import React, { useEffect, useState } from "react";
import ArrowCircleUpRoundedIcon from "@mui/icons-material/ArrowCircleUpRounded";

export const BackToTopButton = () => {
  const [backToTopButton, setBackToTopButton] = useState(false);

  useEffect(() => {
    window.addEventListener("scroll", () => {
      if (window.scrollY > 600) {
        setBackToTopButton(true);
      } else {
        setBackToTopButton(false);
      }
    });
  }, []);

  const scrollToTop = () => {
    window.scrollTo({
      top: 0,
      behavior: "smooth",
    });
  };

  return (
    <div
      style={{
        position: "fixed",
        bottom: "20px",
        right: "50%",
      }}
    >
      {backToTopButton && (
        <ArrowCircleUpRoundedIcon
          onClick={scrollToTop}
          sx={{
            backgroundColor: "white",
            borderRadius: "50%",
            width: 50,
            height: 50,
            '&:hover': {
              cursor: 'pointer'
            }
          }}
        />
      )}
    </div>
  );
};
