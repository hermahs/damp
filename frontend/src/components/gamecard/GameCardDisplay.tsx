import React, { useState } from "react";
import { ButtonBase, Grid, Paper, styled, Typography } from "@mui/material";
import { observer } from "mobx-react";
import { Game } from "../../types";

const Img = styled('img')({
    margin: 'auto',
    display: 'block',
    maxWidth: '120%',
    maxHeight: '200%',
  });

export const GameCardDisplay = observer((props: { game: Game, onClick: () => void }) => {
    const [isFocus, setIsFocus] = useState(false);

   const handleFocusIn = () => {
      setIsFocus(true);
   };
   const handleFocusOut = () => {
      setIsFocus(false);
   };

    const showRating = (text: string) => {
        if (text === "" || text === null) {
            return "";
        }
        const index = text.indexOf('-')

        if (text.substring(index+2, index+4) === "Ne") {
            return "";
        }

        return "⭐ " + text.substring(index+2, index+4) + "%";
    };

    const showImage = (url: string) => {
        if (url === null) {
            return(<Img alt="complex" src="/images/default.png" />)
        }
        return(<Img alt="complex" src={url} />)
    };

    return (
        <div className="Card" onFocus={handleFocusIn}
        onBlur={handleFocusOut}>
        <Paper
            sx={{
                p: 2,
                margin: "auto",
                maxWidth: 500,
                flexGrow: 1,
                backgroundColor: (theme) =>
                theme.palette.mode === "dark" ? "#1A2027" : "#fff",
                marginBottom: 1,
                marginTop: 1,
                boxShadow: 3,
                background: isFocus? "#ebecf0" :null,
            }}
            data-testid={`gameCard-${props.game.name.replace(/\s/g, '')}`}
            onClick={props.onClick}
        >
        <Grid container spacing={4}>
            <Grid item>
            <ButtonBase data-testid={"img" + props.game.appId} sx={{ width: 100, height: 100 }}>
                {showImage(props.game.imagePath)}
            </ButtonBase>
            </Grid>
            <Grid item xs={6} sm container>
            <Grid item xs container direction="column" spacing={2}>
                <Grid item xs>
                <Typography gutterBottom component="div" variant="h5">
                    {props.game.name}
                </Typography>
                <Typography variant="body2" gutterBottom>
                    Published: {props.game.release_date}
                </Typography>
                <Typography variant="body2" gutterBottom>
                    Publisher:{" "}
                    {props.game.publisher !== null ? props.game.publisher[0] : props.game.developer}
                </Typography>
                <Typography variant="body2" gutterBottom>
                    Genre: {props.game.genre !== null ? props.game.genre[0] : "No genre"}
                </Typography>
                </Grid>
            </Grid>
            <Grid item xs="auto">
                <Typography variant="subtitle1" component="div">
                {showRating(props.game.all_reviews)}
                </Typography>
            </Grid>
            </Grid>
        </Grid> 
        </Paper>
        </div>
    );
});
