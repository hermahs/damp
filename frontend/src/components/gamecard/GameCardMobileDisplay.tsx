import React from 'react';
import Card from '@mui/material/Card';
import CardContent from '@mui/material/CardContent';
import CardMedia from '@mui/material/CardMedia';
import Typography from '@mui/material/Typography';
import { CardActionArea, Grid } from '@mui/material';
import { observer } from "mobx-react";
import { Game } from "../../types";


export const GameCardMobileDisplay = observer((props: { game: Game, onClick: () => void }) => {

    const showRating = (text: string) => {
        if (text === "" || text === null) {
            return "";
        }
        const index = text.indexOf('-')
        return "⭐ " + text.substring(index+2, index+4) + "%";
    };

    const showImage = (url: string) => {
        if (url === null) {
            return(<CardMedia alt="complex" src="/images/default.png" component="img" height="140"/>)
        }
        return(<CardMedia alt="complex" src={url} component="img" height="140"/>)
    };


    return (
        <Card sx={{ maxWidth: 345, boxShadow: 3, margin: "auto", marginBottom: 1, marginTop: 1, display: 'block',
                    backgroundColor: (theme) =>theme.palette.mode === "dark" ? "#1A2027" : "#fff", }}
              onClick={props.onClick}>
          <CardActionArea>
          {showImage(props.game.imagePath)}
            <CardContent>
            <Grid sm container>
            <Grid item xs container direction="column">
              <Typography gutterBottom variant="h5" component="div">
                {props.game.name}
              </Typography>
                <Typography variant="body2" gutterBottom color="text.secondary">
                    Published: {props.game.release_date}
                </Typography>
                <Typography variant="body2" gutterBottom color="text.secondary">
                    Publisher:{" "}
                    {props.game.publisher !== null ? props.game.publisher[0] : props.game.developer}
                </Typography>
                <Typography variant="body2" gutterBottom color="text.secondary">
                    Genre: {props.game.genre !== null ? props.game.genre[0] : "No genre"}
                </Typography>
                </Grid>
                <Grid item xs="auto">
                    <Typography variant="subtitle1" component="div">
                        {showRating(props.game.all_reviews)}
                    </Typography>
                </Grid>
                </Grid>
            </CardContent>
          </CardActionArea>
        </Card>
      );
});



