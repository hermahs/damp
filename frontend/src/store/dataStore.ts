import { action, IObservableArray, makeObservable, observable, runInAction } from "mobx";
import { Game, SortType } from "../types";
import { LOAD_GAMES } from "../graphQL";
import { client } from "../util";
import { ApolloError, ObservableQuery } from "@apollo/client";
import { RootStore } from "./rootStore";

interface ISort {
    ascending: boolean;
    type: SortType;
}

export class DataStore {
    data: IObservableArray<Game>;
    pageNumber: number;
    searchString: string = "";
    loading: boolean;
    allFound: boolean = false;
    error: ApolloError | undefined;
    sort: ISort;
    rootStore: RootStore
    query: ObservableQuery = client.watchQuery({query: LOAD_GAMES});

    constructor(rootStore: RootStore) {
        this.pageNumber = 0
        this.loading = false;
        this.error = undefined;
        this.data = observable.array<Game>([]);
        this.sort = {
            ascending: true,
            type: SortType.NONE
        };
        this.rootStore = rootStore;
        makeObservable(this, {
            data: observable,
            loading: observable,
            error: observable,
            allFound: observable,
            searchString: observable,
            getMoreData: action,
            reloadData: action,
            setSort: action,
            setSearchString: action
        });
        runInAction(() => {
            this.getMoreData();
        });
    }

    async reloadData() {
        runInAction(() => {
            this.pageNumber = 0;
            this.data.clear();
            this.allFound = false;
            this.getMoreData();
        });
    }

    setSearchString(s: string) {
        this.searchString = s;
    }

    async getMoreData() {
        const variables = {
            page: this.pageNumber,
            filter: this.rootStore.filterStore.queryFilter,
            sort: this.sort,
            search: this.searchString
        };
        this.error = undefined;
        this.loading = true;
        client.query({
            query: LOAD_GAMES,
            variables
        }).then(
            action("getSucces", ({data}) => {
                this.data.push(...data.games);
                this.loading = false;
                this.pageNumber++;
                if (data.games.length < 15) {
                    this.allFound = true;
                }
            })
        ).catch(action("getError", (error: ApolloError) => {
            console.log(error);
            this.error = error;
        }))
    }

    setSort(type: SortType, ascending: boolean = true) {
        this.sort.ascending = ascending;
        this.sort.type = type;
        this.reloadData();
    }

}