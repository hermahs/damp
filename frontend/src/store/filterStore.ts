import { observable, action, makeObservable, runInAction, computed } from 'mobx';
import { RootStore } from './rootStore';

export interface IFilterData {
    type: string;
    data: number | string[] | number[] | (Date | null)[];
    visualData: string;
}

export interface IFilter {
    name: string;
    color: string;
    description: string;
    data?: IFilterData;
}

interface ReturnData {
    genre?: string[];
    price?: number[];
    releasedate?: (Date | null)[];
    achivements?: number;
    tags?: string[];
}

export class FilterStore {
    activeFilters: Set<IFilter> = new Set<IFilter>();
    allFilters: Set<IFilter> = new Set<IFilter>([
        {name: "Genre", color: "#F2994A", description: "The genre of the game"}, 
        {name: "Price", color: "#F2C94C", description: "The price of the game"}, 
        {name: 'ReleaseDate', color: '#BB6BD9', description: "The release date of the game"}, 
        {name: "Achievements", color: "#2D9CDB", description: "The number of achievements of the game"}, 
        {name: "Tags", color: "#EB5757", description: "The tags of the game"}]);
    queryFilter: ReturnData = observable.object<ReturnData>({});
    rootStore: RootStore;
    selectedFilter: IFilter | undefined;
    showModalInfo: boolean = false;

    constructor(rootStore: RootStore) {
        this.rootStore = rootStore;
        makeObservable(this, {
            activeFilters: observable,
            allFilters: observable,
            queryFilter: observable,
            addFilter: action,
            removeFilter: action,
            resetStore: action,
            resetable: computed
        });
    }

    get resetable(): boolean {
        if (Array.from(this.activeFilters).length > 0) return true;
        return false;
    }

    private addFilterQuery (filter: IFilter) {
        switch (filter.name) {
            case "Genre":
                this.queryFilter.genre = (filter.data!.data as string[]);
                break;
            case "Price":
                this.queryFilter.price = (filter.data!.data as number[]);
                break;
            case "ReleaseDate":
                this.queryFilter.releasedate = (filter.data!.data as (Date | null)[]);
                break;
            case "Achievements":
                this.queryFilter.achivements = (filter.data!.data as number);
                break;
            case "Tags":
                this.queryFilter.tags = (filter.data!.data as string[]);
                break;
        }
    }

    private removeFilterQuery (filter: IFilter) {
        type objectKey = keyof ReturnData;
        delete this.queryFilter[filter.name.toLowerCase() as objectKey];
    }

    private getFromSet(set: Set<IFilter>, name: string): IFilter {
        let elem: IFilter;
        for(let a of Array.from(set)) {
            if (a.name === name) {
                elem = a;
                break;
            }
        }
        set.delete(elem!);
        return elem!;
    }

    addFilter(filter: string, data: IFilterData) {
        const f = this.getFromSet(this.allFilters, filter);
        f.data = data;
        this.addFilterQuery(f);
        this.activeFilters.add(f);
        runInAction(() => {
            this.rootStore.dataStore.reloadData();
        });
    }

    removeFilter(filter: string) {
        const f = this.getFromSet(this.activeFilters, filter);
        this.removeFilterQuery(f);
        this.allFilters.add(f);
        runInAction(() => {
            this.rootStore.dataStore.reloadData();
        });
    }

    selectFilter(filter: IFilter) {
        this.selectedFilter = filter;
        this.showModalInfo = true;
    }

    unSelectFilter() {
        runInAction(() => {
            this.showModalInfo = false;
            this.selectedFilter = undefined;
        });
    }

    resetStore() {
        Array.from(this.activeFilters).forEach(filter => {
            const f = this.getFromSet(this.activeFilters, filter.name);
            this.removeFilterQuery(f);
            this.allFilters.add(f);
        });
    }
}