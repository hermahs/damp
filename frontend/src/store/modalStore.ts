import { ObservableQuery } from "@apollo/client";
import { action, makeObservable, observable, runInAction } from "mobx";
import { GET_SINGLE_GAME } from "../graphQL";
import { Game } from "../types";
import { client } from "../util";
import { RootStore } from "./rootStore";

export class ModalStore {
    selectedGame: number = -1;
    game: Game | undefined;
    showModal: boolean = false;
    updating: boolean = false;
    rootStore: RootStore;
    q: ObservableQuery = client.watchQuery({
        query: GET_SINGLE_GAME
    });

    constructor(rootStore: RootStore) {
        this.rootStore = rootStore;
        makeObservable(this, {
            game: observable,
            showModal: observable,
            updating: observable,
            selectGame: action,
            unSelectGame: action,
            updateGameData: action
        })
    }

    selectGame(appId: number) {
        this.q.refetch({
            appId: appId
        }).then(action("getSuccess", ({data}) => {
            this.game = data.game as Game;
            this.showModal = true;
            this.selectedGame = appId;
        }))
    }

    updateGameData() {
        this.updating = true;
        if (this.selectedGame === -1) return;
        this.q.refetch({
            appId: this.selectedGame
        }).then(action("getSuccess", ({ data }) => {
            this.game = data.game as Game;
            this.updating = false;
        }));
    }

    unSelectGame() {
        runInAction(() => {
            this.showModal = false;
            this.game = undefined;
            this.selectedGame = -1;
        })
    }

}