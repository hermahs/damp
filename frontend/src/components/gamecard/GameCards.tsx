import React, { useEffect, useState } from "react";
import { observer } from 'mobx-react';
import { GameCardDisplay } from "./GameCardDisplay";
import { GameCardMobileDisplay } from "./GameCardMobileDisplay";
import { runInAction } from "mobx";
import { useStores } from '../../hooks';
import { Game } from "../../types";
import { Box, CircularProgress, Typography } from "@mui/material";
import { GameCardModal } from "./GameCardModal";
import { BackToTopButton } from './BackToTopButton';

export const GameCards = observer(() => {
    const { store } = useStores();

    const [mobileDisplay, setMobileDisplay] = useState(false);

    useEffect(() => {
        window.addEventListener("resize", function() {
            if (window.innerWidth < 600) {
                setMobileDisplay(true);
            }
            else {
                setMobileDisplay(false);
            }
        });
    }, []) 

    const handleScroll = (_: Event) => {
        const bottom = window.scrollY + window.innerHeight > document.body.clientHeight - 300;
        if (bottom && !store.dataStore.loading && !store.dataStore.allFound) {
            runInAction(() => {
                store.dataStore.getMoreData();
            })
        }
    }

    useEffect(() => {
        window.addEventListener('scroll', handleScroll, { passive: true });

        if (window.innerWidth < 600) {
            setMobileDisplay(true);
        }
        else {
            setMobileDisplay(false);
        }

        return () => {
            window.removeEventListener('scroll', handleScroll);
        }

    }, [])

    const handleClose = (event: object, reason: string) => {
        store.modalStore.unSelectGame();
    }
    
    return (
        <Box data-testid="gameCardContainer">
            {(store.dataStore.data.length === 0 && !store.dataStore.loading) ? <Typography textAlign={'center'}>No games found :(</Typography> : <></>}
            {!mobileDisplay && store.dataStore.data.map((game: Game) => 
            <GameCardDisplay game={game} key={game.appId} onClick={() => store.modalStore.selectGame(game.appId)}/>)}

            {mobileDisplay && store.dataStore.data.map((game: Game) => 
            <GameCardMobileDisplay game={game} key={game.appId} onClick={() => store.modalStore.selectGame(game.appId)}/>)}
            
            <GameCardModal open={store.modalStore.showModal} onClose={handleClose} />
            {store.dataStore.loading && <Box sx={{mx: 'auto', width: 'fit-content'}}><CircularProgress sx={{mx: 'auto'}}/></Box>}
            <BackToTopButton />
        </Box>
    );
});
            

