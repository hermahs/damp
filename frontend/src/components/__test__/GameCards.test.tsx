import { render, screen, cleanup, getByAltText } from "@testing-library/react";
import { GameCards } from "../gamecard";
import { defaultContext, store } from "../../store";
import { MockedProvider } from "@apollo/client/testing";
import { Provider } from "mobx-react";
import { Game, Comment } from "../../types";
import renderer from "react-test-renderer";
import userEvent from "@testing-library/user-event";

const comment: Comment = {
  name: "the watcher",
  rating: 3,
  comment: "i recommend this game",
};

const game1: Game = {
  name: "Overcooked",
  appId: 379720,
  imagePath: "https://cdn.akamai.steamstatic.com/steam/apps/379720/header.jpg",
  release_date: "May 12, 1962",
  publisher: ["Gutta boys"],
  genre: ["Action"],
  all_reviews: "Very Positive,(3,547),- 84% of the 3,547 user reviews for this game are positive.",
  developer: "Johan",
  game_description: "Lets get cooking, awesome game",
  description: "",
  achievements: 59,
  languages: ["Hindu, Norwegian"],
  price: 42,
  releaseDate: new Date(),
  comments: [comment],
};

const game2: Game = {
  name: "Trackmania",
  appId: 379721,
  imagePath: "https://cdn.akamai.steamstatic.com/steam/apps/379720/header.jpg",
  release_date: "September 17, 1959",
  publisher: ["Kara fra Sahra"],
  genre: ["Strategy"],
  all_reviews: "7 user reviews,- Need more user reviews to generate a score",
  developer: "Han der",
  game_description: "Lets get speedy",
  description: "",
  achievements: 10,
  languages: ["Norsk"],
  price: 45,
  releaseDate: new Date(),
  comments: [],
};

const game3: Game = {
  name: "Elden Ring",
  appId: 379722,
  imagePath: "",
  release_date: "December 6, 1999",
  publisher: ["BÃ¸rge"],
  genre: ["RPG"],
  all_reviews: "",
  developer: "Ludde",
  game_description: "Gotta find the Elden Ring",
  description: "",
  achievements: 0,
  languages: ["English"],
  price: 60,
  releaseDate: new Date(),
  comments: [],
};
store.dataStore.data.push(game1);
store.dataStore.data.push(game2);
store.dataStore.data.push(game3);

afterEach(cleanup);

describe("GameCards test", () => {
  it("renders without crashing, and shows correct games", () => {
    render(
      <Provider {...defaultContext}>
        <MockedProvider addTypename={false}>
          <GameCards />
        </MockedProvider>
      </Provider>
    );

    screen.getByText(/overcooked/i);
    screen.getByText(/May 12, 1962/i);
    screen.getByText(/Action/i);
    screen.getByText(/84%/i);

    screen.getByText(/elden ring/i);
    screen.getByText(/trackmania/i);
  });


  it("game card modal test", () => {
    render(
      <Provider {...defaultContext}>
        <MockedProvider addTypename={false}>
          <GameCards />
        </MockedProvider>
      </Provider>
    );

    userEvent.click(screen.getByText(/overcooked/i));
    //screen.getByText(/description/i);
  });  






























});
/*
describe("Filter snapshot test", () => {
  it("render filter component", () => {
    const tree = renderer
      .create(
        <Provider {...defaultContext}>
          <MockedProvider addTypename={false}>
            <GameCards />
          </MockedProvider>
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
}); */
