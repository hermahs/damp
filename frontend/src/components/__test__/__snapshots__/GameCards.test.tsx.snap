// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter snapshot test render filter component 1`] = `
<div
  className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-1vme478-MuiGrid-root"
  data-testid="gameCardContainer"
>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-10 MuiGrid-grid-sm-6 MuiGrid-grid-md-4 css-bg6aee-MuiGrid-root"
  >
    <div
      className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-133i4ok-MuiPaper-root-MuiCard-root"
      onClick={[Function]}
    >
      <button
        className="MuiButtonBase-root MuiCardActionArea-root css-1pz8km4-MuiButtonBase-root-MuiCardActionArea-root"
        disabled={false}
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <img
          alt="complex"
          className="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-o69gx8-MuiCardMedia-root"
          height="140"
          src="https://cdn.akamai.steamstatic.com/steam/apps/379720/header.jpg"
        />
        <div
          className="MuiCardContent-root css-16d4f6n-MuiCardContent-root"
        >
          <div
            className="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column css-1y7as2u-MuiGrid-root"
          >
            <div
              className="MuiGrid-root MuiGrid-container MuiGrid-item css-9pwih8-MuiGrid-root"
            >
              <div
                className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8 css-vok43g-MuiGrid-root"
              >
                <p
                  className="MuiTypography-root MuiTypography-body1 css-mvvmou-MuiTypography-root"
                >
                  Overcooked
                </p>
              </div>
              <div
                className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 css-1equabv-MuiGrid-root"
              >
                <div
                  className="MuiTypography-root MuiTypography-subtitle1 css-10wpov9-MuiTypography-root"
                />
              </div>
            </div>
            <div
              className="MuiGrid-root MuiGrid-item css-4pkl32-MuiGrid-root"
            >
              <div
                className="MuiBox-root css-9jay18"
              >
                <p
                  className="MuiTypography-root MuiTypography-body2 MuiTypography-gutterBottom css-186q1z4-MuiTypography-root"
                >
                  May 12, 1962
                </p>
                <p
                  className="MuiTypography-root MuiTypography-body2 MuiTypography-gutterBottom css-186q1z4-MuiTypography-root"
                >
                  42$
                </p>
              </div>
            </div>
            <div
              className="MuiGrid-root MuiGrid-item css-1nyge9p-MuiGrid-root"
            >
              <div
                className="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-filledDefault css-8649g4-MuiChip-root"
                onKeyDown={[Function]}
                onKeyUp={[Function]}
              >
                <span
                  className="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
                >
                  Action
                </span>
              </div>
            </div>
          </div>
        </div>
        <span
          className="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
        />
      </button>
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-10 MuiGrid-grid-sm-6 MuiGrid-grid-md-4 css-bg6aee-MuiGrid-root"
  >
    <div
      className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-133i4ok-MuiPaper-root-MuiCard-root"
      onClick={[Function]}
    >
      <button
        className="MuiButtonBase-root MuiCardActionArea-root css-1pz8km4-MuiButtonBase-root-MuiCardActionArea-root"
        disabled={false}
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <img
          alt="complex"
          className="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-o69gx8-MuiCardMedia-root"
          height="140"
          src="https://cdn.akamai.steamstatic.com/steam/apps/379720/header.jpg"
        />
        <div
          className="MuiCardContent-root css-16d4f6n-MuiCardContent-root"
        >
          <div
            className="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column css-1y7as2u-MuiGrid-root"
          >
            <div
              className="MuiGrid-root MuiGrid-container MuiGrid-item css-9pwih8-MuiGrid-root"
            >
              <div
                className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8 css-vok43g-MuiGrid-root"
              >
                <p
                  className="MuiTypography-root MuiTypography-body1 css-mvvmou-MuiTypography-root"
                >
                  Trackmania
                </p>
              </div>
              <div
                className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 css-1equabv-MuiGrid-root"
              >
                <div
                  className="MuiTypography-root MuiTypography-subtitle1 css-10wpov9-MuiTypography-root"
                />
              </div>
            </div>
            <div
              className="MuiGrid-root MuiGrid-item css-4pkl32-MuiGrid-root"
            >
              <div
                className="MuiBox-root css-9jay18"
              >
                <p
                  className="MuiTypography-root MuiTypography-body2 MuiTypography-gutterBottom css-186q1z4-MuiTypography-root"
                >
                  September 17, 1959
                </p>
                <p
                  className="MuiTypography-root MuiTypography-body2 MuiTypography-gutterBottom css-186q1z4-MuiTypography-root"
                >
                  45$
                </p>
              </div>
            </div>
            <div
              className="MuiGrid-root MuiGrid-item css-1nyge9p-MuiGrid-root"
            >
              <div
                className="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-filledDefault css-158wpwh-MuiChip-root"
                onKeyDown={[Function]}
                onKeyUp={[Function]}
              >
                <span
                  className="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
                >
                  Strategy
                </span>
              </div>
            </div>
          </div>
        </div>
        <span
          className="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
        />
      </button>
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-10 MuiGrid-grid-sm-6 MuiGrid-grid-md-4 css-bg6aee-MuiGrid-root"
  >
    <div
      className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-133i4ok-MuiPaper-root-MuiCard-root"
      onClick={[Function]}
    >
      <button
        className="MuiButtonBase-root MuiCardActionArea-root css-1pz8km4-MuiButtonBase-root-MuiCardActionArea-root"
        disabled={false}
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <img
          alt="complex"
          className="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-o69gx8-MuiCardMedia-root"
          height="140"
          src="https://cdn.akamai.steamstatic.com/steam/apps/379720/header.jpg"
        />
        <div
          className="MuiCardContent-root css-16d4f6n-MuiCardContent-root"
        >
          <div
            className="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column css-1y7as2u-MuiGrid-root"
          >
            <div
              className="MuiGrid-root MuiGrid-container MuiGrid-item css-9pwih8-MuiGrid-root"
            >
              <div
                className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8 css-vok43g-MuiGrid-root"
              >
                <p
                  className="MuiTypography-root MuiTypography-body1 css-mvvmou-MuiTypography-root"
                >
                  Elden Ring
                </p>
              </div>
              <div
                className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 css-1equabv-MuiGrid-root"
              >
                <div
                  className="MuiTypography-root MuiTypography-subtitle1 css-10wpov9-MuiTypography-root"
                />
              </div>
            </div>
            <div
              className="MuiGrid-root MuiGrid-item css-4pkl32-MuiGrid-root"
            >
              <div
                className="MuiBox-root css-9jay18"
              >
                <p
                  className="MuiTypography-root MuiTypography-body2 MuiTypography-gutterBottom css-186q1z4-MuiTypography-root"
                >
                  December 6, 1999
                </p>
                <p
                  className="MuiTypography-root MuiTypography-body2 MuiTypography-gutterBottom css-186q1z4-MuiTypography-root"
                >
                  60$
                </p>
              </div>
            </div>
            <div
              className="MuiGrid-root MuiGrid-item css-1nyge9p-MuiGrid-root"
            >
              <div
                className="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-filledDefault css-sbrmq3-MuiChip-root"
                onKeyDown={[Function]}
                onKeyUp={[Function]}
              >
                <span
                  className="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
                >
                  RPG
                </span>
              </div>
            </div>
          </div>
        </div>
        <span
          className="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
        />
      </button>
    </div>
  </div>
  <div
    className="MuiBox-root css-3rgs8z"
  >
    <span
      className="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-5106e7-MuiCircularProgress-root"
      role="progressbar"
      style={
        Object {
          "height": 40,
          "width": 40,
        }
      }
    >
      <svg
        className="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
        viewBox="22 22 44 44"
      >
        <circle
          className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
          cx={44}
          cy={44}
          fill="none"
          r={20.2}
          strokeWidth={3.6}
          style={Object {}}
        />
      </svg>
    </span>
  </div>
  <div
    style={
      Object {
        "bottom": "20px",
        "position": "fixed",
        "right": "50%",
      }
    }
  />
</div>
`;
